File: internal/content/testdata/bench/go/max-depth-binary-tree.go
func max_depth(root *TreeNode) int {
	if root == nil {
		return 0
	}
	left := max_depth(root.Left)
	right := max_depth(root.Right)
	if left > right {
		return 1 + left
	}
	return 1 + right
}

File: internal/content/testdata/bench/rust/max-depth-binary-tree.rs
use std::rc::Rc;
use std::cell::RefCell;

fn max_depth(root: Option<Rc<RefCell<TreeNode>>>) -> i64 {
    match root {
        None => 0,
        Some(node) => {
            let n = node.borrow();
            1 + max_depth(n.left.clone()).max(max_depth(n.right.clone()))
        }
    }
}
